WordPress/gutenberg
# BottomSheet listens to native bridge for closed modals on Android

Fixes: https://github.com/wordpress-mobile/WordPress-Android/issues/10227

### Related PRs:
GB-Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2011
WordPress-Android: https://github.com/wordpress-mobile/WordPress-Android/pull/10383

### Description
When the Gutenberg editor on Android is contained in a `Fragment`, there are cases on device orientation change where the native Android `ReactInstanceManager` will dismiss open Modals (`Dialog` in Android Native code) in order to avoid a crash. 

This PR allows our Modal utilizing `BottomSheet` component to subscribe to messages from the native Android bridge that notify when a Modal is dismissed from native Android, and we'll perform the necessary React Native state updates accordingly by calling the `BottomSheet` `onClose` method. Failure to call `onClose` method in this instance results in non functioning buttons for opening modals, because React Native state thinks they are still open, hence this change. 

### How has this been tested?



Tested in WordPress Android app, as well as in the Android/iOS example apps. See steps and screenshot below.

Testing steps (same apply for opening the inserter Modal, or clicking insert Image/Video to Open a modal)
1. Add an Image block and add an image from device or media library.
2. While the image block is selected, press the "gear" button to open up the Image Settings.
3. Rotate the Device.
4. Click the gear button again, the Modal should reopen as normal.
5. Rotate the Device again => the app should behave normally without crashing. 

## Screenshots 
![not-crashing-rotate-modal-android](https://user-images.githubusercontent.com/1103838/76661908-d1770180-6539-11ea-9613-06c43a124d67.gif)


## Types of changes
Bug fix 

## Checklist:
